lib/phobos/listener.rb

```diff
@@ -42,12 +42,14 @@ def initialize(handler:, group_id:, topic:, min_bytes: nil, max_wait_time: nil,
 
     def start
       @signal_to_stop = false
+      error = nil
 
       start_listener
 
       begin
         start_consumer_loop
-      rescue Kafka::ProcessingError, Phobos::AbortError
+      rescue Kafka::ProcessingError, Phobos::AbortError => e
+        error = e if e.is_a?(Kafka::ProcessingError)
```

Comment:
`raise e if e.is_a?(Kafka::ProcessingError)`

```ruby
begin
   puts "begin block"
   raise 'blue'
rescue Exception
   puts "error block";
   raise 'green'
ensure
  puts "always block";
end
```
You can raise error in rescue block. Ensure block runs always.
You can avoid the error variable entirely. Just wondering if I am missing something.

Reply:
I just realised the structure is little different after taking a look at the listener.rb file. However, raising an error in rescue works as expected.

```ruby
def method1
  begin
    puts 'begin block'
    raise 'BeginError'
  rescue Exception => e
    puts 'exception block'
    raise e
  end
ensure
  puts 'method ensure'
end
```
Output:
> begin block
> exception block
> method ensure
> Traceback (most recent call last):
>         5: from /Users/harsha.subraveti/.rvm/rubies/ruby-3.0.0/bin/irb:23:in `<main'`
>         4: from /Users/harsha.subraveti/.rvm/rubies/ruby-3.0.0/bin/irb:23:in `load'`
>         3: from /Users/harsha.subraveti/.rvm/rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/irb-1.3.0/exe/irb:11:in `<top (required)'
>         2: from (irb):12:in `<main'
>         1: from (irb):4:in `method1'
> RuntimeError (BeginError)